src/main/java/snowy/Deadline.java

```diff
@@ -0,0 +1,39 @@
+package snowy;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+/**
+ * Represents a task that have a deadline to be completed by.
+ * This class holds the deadline as a LocalDate.
+ */
+public class Deadline extends Task {
+    private LocalDate date;
```

Comment:
Maybe consider changing the name to the endDate or deadline? It would be clearer.